sympy/functions/elementary/exponential.py

```diff
@@ -768,7 +768,7 @@ def _eval_expand_log(self, deep=True, **hints):
                 else:
                     return unpolarify(e) * a
         elif isinstance(arg, Product):
-            if arg.function.is_positive:
+            if force or arg.function.is_positive:
                 return Sum(log(arg.function), *arg.limits)
```

Comment:
Also here is it sufficient that the function is positive?

Reply:
If all elements of the product are positive, then the product is positive, therefore it should be possible to expand the log.

Reply:
It's the infinite Sum/Product case I wonder about.

Reply:
It would be nice if this worked as well, but that may be another PR:

```
expr = log(Product(x**i, (i, 1, N)))
expr.rewrite(Sum).expand()
```

Btw, this works now: `expr.rewrite(Sum).doit().expand()`

I think the problem is that `Sum` does not have a (good enough) `is_extended_real`-handler, so probably another PR...